### PR TITLE
fixes deprecated warning introduced at #26198

### PR DESCRIPTION
**Pull request by ohadlevy**

### CODE REVIEW

File: app/views/discovered_hosts/_discovered_hosts_list.html.erb [hunk 1]
```diff
@@ -1,8 +1,7 @@
-<%= javascript "host_checkbox" %>
```
Comment (ohadlevy) [inline app/views/discovered_hosts/_discovered_hosts_list.html.erb:1]: this is the actual fix, all the other changes are to avoid the deprecation warnings.

**Comment by lzap** [comment]
I still see them :-(

```
Error:
DiscoveredHostsTest::using Create Host link::with a Hostgroup selected#test_0001_it passes it on:
ActionView::Template::Error: DEPRECATION WARNING: You are using a deprecated behavior, it will be removed in version 1.24, passing :confirm parameter in html_options to link_to is deprecated. Please use `data: {confirm: "confirmation message?"}` instead. (called from display_link_if_authorized at /home/lzap/work/foreman/app/helpers/application_helper.rb:145)
    app/helpers/application_helper.rb:145:in `display_link_if_authorized'
    app/helpers/application_helper.rb:136:in `display_delete_if_authorized'
    /home/lzap/work/foreman_discovery/app/views/discovered_hosts/_discovered_hosts_list.html.erb:39:in `block in __home_lzap_work_foreman_discovery_app_views_discovered_hosts__discovered_hosts_list_html_erb___102052519190562205_46912635869800'
    /home/lzap/work/foreman_discovery/app/views/discovered_hosts/_discovered_hosts_list.html.erb:21:in `__home_lzap_work_foreman_discovery_app_views_discovered_hosts__discovered_hosts_list_html_erb___102052519190562205_46912635869800'
    /home/lzap/work/foreman_discovery/app/views/discovered_hosts/index.html.erb:2:in `__home_lzap_work_foreman_discovery_app_views_discovered_hosts_index_html_erb__2852598023806942253_46912635711560'
    app/controllers/concerns/foreman/controller/timezone.rb:10:in `set_timezone'
    app/models/concerns/foreman/thread_session.rb:32:in `clear_thread'
    app/controllers/concerns/foreman/controller/topbar_sweeper.rb:12:in `set_topbar_sweeper_controller'
    lib/foreman/middleware/telemetry.rb:10:in `call'
    lib/foreman/middleware/catch_json_parse_errors.rb:9:in `call'
    lib/foreman/middleware/logging_context_session.rb:22:in `call'
    lib/foreman/middleware/logging_context_request.rb:11:in `call'
```

Edit: looks like a different bug.

**Comment by ohadlevy** [comment]
these are plain foreman depreciation... (ruby) want me to fix them for you
too? :)

**Comment by lzap** [comment]
Nope thanks I am pulling this into my other PR where I aim to fix them all, thanks!

**Pull request closed without merging**